core/src/test/java/jenkins/security/apitoken/ApiTokenStatsTest.java

```diff
@@ -0,0 +1,224 @@
+
+import java.lang.reflect.Field;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ApiTokenPropertyConfiguration.class)
+public class ApiTokenStatsTest {
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+    
+    @Before
+    public void prepareConfig() throws Exception {
+        // to separate completely the class under test from its environment
+        ApiTokenPropertyConfiguration mockConfig = Mockito.mock(ApiTokenPropertyConfiguration.class);
+        Mockito.when(mockConfig.isUsageStatisticsEnabled()).thenReturn(true);
+        
+        PowerMockito.mockStatic(ApiTokenPropertyConfiguration.class);
+        PowerMockito.when(ApiTokenPropertyConfiguration.class, "get").thenReturn(mockConfig);
+    }
+    
+    @Test
+    public void regularUsage() throws Exception {
+        final String ID_1 = UUID.randomUUID().toString();
+        final String ID_2 = "other-uuid";
+        
+        { // empty stats can be saved
+            ApiTokenStats tokenStats = new ApiTokenStats();
+            tokenStats.setParent(tmp.getRoot());
+            
+            // can remove an id that does not exist
+            tokenStats.removeId(ID_1);
+            
+            tokenStats.save();
+        }
+        
+        { // and then loaded, empty stats is empty
+            ApiTokenStats tokenStats = ApiTokenStats.load(tmp.getRoot());
+            assertNotNull(tokenStats);
+            
+            ApiTokenStats.SingleTokenStats stats = tokenStats.findTokenStatsById(ID_1);
+            assertEquals(0, stats.getUseCounter());
+            assertNull(stats.getLastUseDate());
+            assertEquals(0L, stats.getNumDaysUse());
+        }
+        
+        Date lastUsage;
+        { // then re-notify the same token
+            ApiTokenStats tokenStats = ApiTokenStats.load(tmp.getRoot());
+            
+            ApiTokenStats.SingleTokenStats stats = tokenStats.updateUsageForId(ID_1);
+            assertEquals(1, stats.getUseCounter());
+            
+            lastUsage = stats.getLastUseDate();
+            assertNotNull(lastUsage);
+            // to avoid flaky test in case the test is run at midnight, normally it's 0 
+            assertTrue(stats.getNumDaysUse() <= 1);
+        }
+        
+        // to enforce a difference in the lastUseDate
+        Thread.sleep(10);
+        
+        { // then re-notify the same token
+            ApiTokenStats tokenStats = ApiTokenStats.load(tmp.getRoot());
+            
+            ApiTokenStats.SingleTokenStats stats = tokenStats.updateUsageForId(ID_1);
+            assertEquals(2, stats.getUseCounter());
+            assertTrue(lastUsage.before(stats.getLastUseDate()));
```

Comment:
NIT: never use assertTrue or false for comparasions that can be better represented as `assertThat(target, lessThan(comparison))`
WIthout the code as is should this fail you have lost information as to the lastUsage and the stats. with the latter if it fails you are shown the two values